src/main/java/uk/org/tombolo/ImporterSpecificationRunner.java

```diff
@@ -0,0 +1,120 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.org.tombolo.core.Attribute;
+import uk.org.tombolo.core.Datasource;
+import uk.org.tombolo.core.SubjectType;
+import uk.org.tombolo.importer.Importer;
+
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Get a list of datasources for an importer
+ */
+public class ImporterSpecificationRunner extends AbstractRunner {
+    private static final Logger log = LoggerFactory.getLogger(ImporterSpecificationRunner.class);
+    private static final ImporterSpecificationRunner runner = new ImporterSpecificationRunner();
+
+    public static void main(String[] args) throws Exception {
+
+        boolean subjects = Boolean.parseBoolean(args[1]);
+        boolean attribute = Boolean.parseBoolean(args[2]);
+        boolean provider = Boolean.parseBoolean(args[3]);
+        String importerClass = args[0];
+        StringBuilder info = new StringBuilder("\n\n");
+
+        if (importerClass.equals("None")) {
+            CatalogueExportRunner cer = new CatalogueExportRunner();
+            List<Class<? extends Importer>> dcImporters = cer.getImporterClasses();
+            info.append("List of Digital Connector Importers: \n\n");
+            dcImporters.forEach(i -> info.append(i.getCanonicalName()).append("\n"));
+            log.info(info.toString());
+            return;
+        }
+
+        Importer importer = runner.getImporter(importerClass);
+        List<String> datasources = runner.getDatasourceIds(importer);
+        
+        for (String d : datasources) {
+            info.setLength(0);
+            Datasource dSource = runner.getDatasource(d, importer);
+            if (null != dSource.getDatasourceSpec()) {
+                info.append("\n\nDatasource Id: ")
+                    .append(dSource.getDatasourceSpec().getId())
+                    .append("\nImporter Class: ")
+                    .append(dSource.getDatasourceSpec().getImporterClass().getCanonicalName())
+                    .append("\nData Url: ")
+                    .append(dSource.getDatasourceSpec().getUrl()).append("\n");
+                
+                if (provider) {
+                    info.append("Provider: ")
+                        .append(runner.getProvider(importer)).append("\n");
+                }       
+
+                if (subjects) {
+                    info.append("SubjectTypes: ")
+                        .append(runner.getSubjectType(dSource).toString()).append("\n");
+
+                }
+                if (attribute) {
+                    info.append("Timed Value Attributes: ")
+                        .append(runner.getTimedValueAttributes(dSource).toString())
+                        .append("\nFixed Value Attributes: ")
+                        .append(runner.getFixedValueAttributes(dSource).toString())
+                        .append("\n");
+                }
+                log.info(info.toString());
```

Comment:
Logging here shows too many `2018-03-14 11:18:29.239 [main] INFO  u.o.t.ImporterSpecificationRunner - ` if multiple datasources are available, that make it look ugly. It might be better to log after the loop.

Reply:
i could remove this, but print after the loop could take some time and i think it would be good if user would see some results so that user would know that it is running and second have something to work with

Reply:
It's ok, it was my opinion on a couple of them that I tested.